### DISCUSSION TITLE
invalid command name "customlib::SetBaseRoot"

### DISCUSSION THREAD

**Discussion by SiaCivil** [Anura3D/Anura3D_OpenSource#134] [General]
I have recently started using Anura3D and followed the software installation and tutorial diligently. However, I encountered an issue that I would like to bring to your attention.

I am using GID version 16.1.6d in conjunction with the latest release of Anura3D_2023. Following the tutorial examples, I attempted to create a model and generate Anura3D files. Unfortunately, each time I tried, I received the following error message:
```
invalid command name "customlib::SetBaseRoot"
invalid command name "customlib::SetBaseRoot"
    while executing
"customlib::SetBaseRoot $root"
    (procedure "Anura3D::WriteCalculationFile_CPS" line 8)
    invoked from within
"Anura3D::WriteCalculationFile_CPS $CPS_name"
    (procedure "Anura3D::Calculate" line 20)
    invoked from within
"Anura3D::Calculate"
    (menu invoke)
```
Despite receiving a message indicating that the process had finished successfully, the generated ".GOM" and ".CPS_001" files turned out to be empty. While I'm not entirely certain about the root cause of this problem, I have implemented a temporary solution by removing the following line of code:

`customlib::SetBaseRoot $root`
This line was removed from the following files:

GiD_Problemtype/Anura3D_2023.gid/scripts/createGOM.tcl
GiD_Problemtype/Anura3D_2023.gid/scripts/createCPS.tcl

After making this adjustment, I was able to generate the Anura3D files successfully. However, I am concerned about the potential implications of removing this line of code and whether it might affect the input files in any way.

I would greatly appreciate your assistance in identifying the root cause of this issue and whether my temporary solution is advisable. Thank you for your ongoing support and for the incredible work you do.

**Comment by jacobosaa** [comment]
Hi Sia! 
Hoping that this message finds you well, it hapens that this message also appears to me when i'm trying to calculate. as you remove those lines in the createGOM and createCPS files, can you proceed after like usual? I mean, does work normally or now it has another kind of flaw?
I will remove those lines right now and I'll tell you how it goes.

Greetings from Chile!

**Comment by SiaCivil** [comment]
I'm glad to hear that I'm not the only one experiencing this issue. To answer your question, after removing those lines from the createGOM and createCPS files, I was able to proceed as usual and successfully generate the Anura3D files. As of now, everything seems to be working without any problems.

However, I'd advise caution since I'm not entirely sure about the implications of removing that specific line of code. It's possible that it might affect the input files or the generated results in some way that I haven't observed yet. For now, it's a temporary solution that has allowed me to move forward with my work.

Please do let me know if you encounter any other issues after making the changes. It would be helpful to compare notes and see if there are any other potential solutions or workarounds.

Best wishes,
Greetings from Japan.

**Reply by jacobosaa** [reply]
I'm glad to hear from you. As I mentioned in my last message, I removed those lines from the code and proceeded to run the model. I'd like to explain what I'm modeling so that we have some context. I'm modeling the column collapse example from the tutorial manual, using two constitutive models. The first is the Mohr-Coulomb model for a fully saturated column, and the second is the Bingham model. I'm comparing the behavior of both constitutive models for different solids content. My main goal is to identify a range of solids content for which both models work similarly. This range could be proposed for modeling tailings as the material for the column.

The issue is that when I use the Bingham model, I encounter this message. I'm not sure what's causing it, but perhaps you've encountered it before and can offer some assistance. :)

Once again, I'm truly grateful for your response. I hope everything is going well for you.

Best wishes.

![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/b62273a0-27ec-417e-be25-b90edb29fbb4)

**Comment by gdicarluccio** [comment]
Hi all, we found that the error appears when using the developer version of GiD (that ends with a "d", e.g. GiD 16.1.6d).
We recommend downloading the official one.

**Reply by Abbki** [reply]
Hi @gdicarluccio 
I had the same problem when using GID 16.0.7. The newest GID official version.
To prevent the error, I used an older version, GID 16.0.2, the version that was used in the tutorial manual 2023.

**Reply by CH-max92** [reply]
You are right，Abbki. The newest GiD software (16.0.7) seems to not appear to be compatible with Anura3D.The use of the GiD ver.16.0.2 might solve the problem. Thanks for you sharing.

**Reply by gdicarluccio** [reply]
Hi all, we fixed the bug. The problemtype is now updated.

**Reply by brunohoo** [reply]
Hi there, new to Anura3D here and doing the first tutorial exercise. Found the same error messages as related in this thread using GiD 16.0.9 – also noticed that the GOM file that was generated in the A3D creation was empty. Went back to GiD 16.0.2 and it worked. Had to do the exercise all over again, as some functions from GiD 16.0.9 were not available in 16.0.2. Happy that it worked out :)
